JavaScript/JavaScript.sublime-syntax

```diff
@@ -1363,8 +1370,24 @@ contexts:
     - match: (?={{class_element_name}})
       push: class-element
 
+  static-block:
+    - match: static{{identifier_break}}
+      scope: storage.modifier.js
+      set:
+        - match: \{
+          scope: punctuation.section.block.begin.js
+          set:
+            - meta_scope: meta.block.js
+            - match: \}
+              scope: punctuation.section.block.end.js
+              pop: true
+            - include: statements
+        - match: (?=\S)
+          fail: static-block
+
   class-element-modifiers:
     - match: (?={{modifier}})
```

Comment:
The way this context is used, would allow to use and empty pattern `- match: ''` instead.

Reply:
In order to do that, we'd have to consume and scope static in the rule in `class-body-contents` that branches to `static-block` and `class-element-modifiers`. But that would break in this case:

```js
class C {
  static() {}
}
```

Reply:
Not sure about that. 

1. I don't see anything changing when turning that pattern into an empty one.
2. `class-element-modifiers` is pushed on stack only by the branch_point `static-block` and the following pattern which both contain `static`. So this pattern must match anyway. Otherwise you'd never pop this context off stack. The first branch `class-element-modifier` consumes `{{modifier}}`. If it fails `static` will be matched as method declaration. But this happens with the current pattern as well.

Reply:
Maybe I'm misunderstanding. Can you post the revised `static-block`?

Reply:
The only thing I did is ...

```
  class-element-modifiers:
    - match: ''
      pop: true
      branch_point: class-element-modifier
      branch:
        - class-element-modifier
        - class-element
```

Java contains several such contexts, which are pushed/branched onto stack by a lookahead and immediately need to hand over treatment to other ones, like this one. It's just the lookahead which is technically not needed in this situation, if I don't miss something.

Reply:
Ah, I thought you were talking about the `static-block` context. I've changed `class-element-modifiers` accordingly.